docs/proposals-accepted/202209-receive-tenant-external-labels.md

```diff
@@ -0,0 +1,52 @@
+---
+type: proposal
+title: Allow specifying tenant-specific external labels in Receivers
+status: accepted
+owner: haanhvu
+menu: proposals-accepted
+---
+
+## 1 Related links/tickets
+
+* https://github.com/thanos-io/thanos/issues/5434
+
+## 2 What problem(s) are we trying to solve?
+
+We want to add arbitrary external labels to each tenant in Thanos Receiver.
+
+## 3 How is it going to be helpful?
+
+This new functionality is helpful in many use cases. Two typical ones are:
+* Compact tenant blocks based on tenants’ external labels
+* Query data from tenants which share the same external labels
+
+## 4 Pitfalls of the current solution
+
+Currently, we can only add external labels to Receivers. Each tenant in a Receiver only gets one external label, which is the tenant ID. This is not flexible enough to do complex compact and query works at the tenant level.
+
+## 5 What are the possible solution approaches?
+
+We’re seeing two solution approaches:
+* Specify the tenants for each external label in Receiver: Since we already have the functionality of adding external labels to Receiver, is there a way to specify which tenants each external label goes to? Because many tenants may share the same external labels, this may be a straightforward way for users.
```

Comment:
Maybe if you have questions, you can organize them in an open questions section below?
Right now it feels like this is actual solution to the problem, and an "answer" rather than a question 🙂

Reply:
Yeah that's not a question anymore. I answered it myself after reading the code :D